### PR TITLE
Created functions to copy campaign, adset or ad

### PR DESCRIPTION
**Pull request by ArbenKqiku**
Hi @daroczig,

I have created 3 functions:
- A function to copy campaigns: fbad_copy_campaign(). **Path**: R/fb_copy_campaign.R
- A function to copy adsets: fbad_copy_adset(). **Path**: R/fb_copy_adset.R
- A function to copy campaigns: fbad_copy_ad(). **Path**: R/fb_copy_ad.R

I extensively tested these 3 functions. You can find the tests that I have run in the folder "tests".

Do you need anything else? When do you think you can accept my pull request?

Best,
Arben

### CODE REVIEW

**Comment by ArbenKqiku** [comment]
Hi @daroczig, did you have time to look at my pull request? Thanks!

File: R/fb_copy_ad.R [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    
+    # check if ad set id actually exists
+    list_of_ads = fbad_list_ad(fbacc)
+    
+    if(!ad_id %in% list_of_ads$id){
+        stop('This ad id does not exists. Please provide a valid ad id.')
+    }
+    
+    # check if adset id actually exists
```
Comment (daroczig) [inline R/fb_copy_ad.R:34]:
```suggestion
    # check if ad id actually exists
```

File: R/fb_copy_ad.R [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                            rename_suffix = NULL, ...) {
+    
+    fbacc <- fbad_check_fbacc()
+    
+    # ad id missing
+    if (missing(ad_id)){
+        stop('Argument missing. An ad id is required.')
+    }
+    
+    # check if ad set id actually exists
```
Comment (daroczig) [inline R/fb_copy_ad.R]:
```suggestion
    # check if ad id actually exists
```

File: R/fb_copy_adset.R [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#' Create a copy of an ad set. You can copy a maximum of 3 entities between ad set and ads.
+#' @inheritParams fbad_request
+#' @param adset_id (numeric string or integer) id adset you want to create a copy of
+#' @param campaign_id (numeric string or integer) Single ID of a campaign to make parent of the copy. Ignore if you want to keep the copy under the original campaign parent
+#' @param start_time (datetime) The start time of the ad set. If not set, the copied adset will inherit the start time from the original set
+#' @param end_time (datetime) The end time of the ad set
+#' @param deep_copy (boolean) Default value: false. Whether to copy all the child ads
+#' @param status_option (string) enum {ACTIVE, PAUSED, INHERITED_FROM_SOURCE}
+#' @param rename_strategy (string) enum {DEEP_RENAME, ONLY_TOP_LEVEL_RENAME, NO_RENAME}
+#' @param rename_prefix (string) a prefix to copy names. Defaults to null if not provided
```
Comment (daroczig) [inline R/fb_copy_adset.R]: no need to copy/paste the shared parameter description -- you could use 

```
#' @inheritParms fbad_copy_ad
```

See https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html

File: fbRads-Arben.Rproj [hunk 4]
```diff
@@ -0,0 +1,17 @@
+Version: 1.0
```
Comment (daroczig) [inline fbRads-Arben.Rproj:1]: Can you please remove this file from the PR along with the `.Rhistory` and your `.httr-oauth`?
Regarding the latter, you might need to reset your credentials as well, as the content of that file become public.

File: tests/test_fb_copy_ad.R [hunk 5]
```diff
@@ -0,0 +1,339 @@
+library(testthat)
```
Comment (daroczig) [inline tests/test_fb_copy_ad.R]: Although I appreciate the effort to test these, but without a token and actual ids (and names), this approach will not really work -- so I suggest removing all these test files from the PR.

**Comment by daroczig** [comment]
@ArbenKqiku -- sorry for the long turn-around time and thanks a lot for the PR, highly appreciated!

I've left a couple of comments -- can you please take a look?

**Pull request closed without merging**